coresimd/powerpc64/vsx.rs

```diff
@@ -216,3 +222,75 @@ impl_from_bits_!(
     vector_bool_long,
     vector_double
 );
+
+mod sealed {
+
+    use super::*;
+
+    pub trait VectorPermDI {
+        unsafe fn vec_xxpermdi(self, b: Self, dm: u8) -> Self;
+    }
+
+    #[inline]
+    #[target_feature(enable = "vsx")]
+    #[cfg_attr(test, assert_instr(xxpermdi, dm = 0x0))]
+    unsafe fn xxpermdi(a: i64x2, b: i64x2, dm: u8) -> i64x2 {
```

Comment:
I think this is missing a `#[rustc_args_required_const(2)]` to require that the user passes a compile-time constant to `dm`.

Reply:
That won't work with the trait approach :/ 
crap

Reply:
Maybe make this function and the trait method impls `#[inline(always)]` ?